diff --git a/Makefile b/Makefile
index 62d640bbf..387b6a109 100755
--- a/Makefile
+++ b/Makefile
@@ -312,6 +312,14 @@ u2plus_swapply:
 	@$(MAKE) -C target/u2plus/nios/ultimate
 	nios2-download -g target/u2plus/nios/ultimate/result/ultimate.elf
 
+u64_swapply:
+	@$(MAKE) -C tools
+	@$(MAKE) -C software/nios_solo_bsp
+	@$(MAKE) -C software/nios_appl_bsp
+	@$(MAKE) -C target/libs/nios2/lwip
+	@$(MAKE) -C target/u64/nios2/ultimate
+	nios2-download -g target/u64/nios2/ultimate/result/ultimate.elf
+
 u2_swonly:
 	@$(MAKE) -C tools
 	@$(MAKE) -C target/u2/microblaze/mb_lwip
